### DISCUSSION TITLE
能接收群消息但是发不出消息

### DISCUSSION THREAD

**Discussion by 1298301390** [simple-robot/Discussions#174] [Q&A]
### 涉及的编程语言

Java

### 涉及的组件库

mirai (QQ)

### 疑问描述

能接受消息，但是无法恢复消息，重启也没用。只能用qr_code的手表扫码登录，其他的都提示code=45 禁止登录，版本不是最新的。

**Comment by ForteScarlet** [comment]
首先确保消息发送没有报错，其次尝试通过私聊回复、仅回复简单字符（例如回复一个 `1`）等。
如果发送消息没有产生异常则大概率是风控导致的（尤其是手表协议）。参考：
- #158
- https://github.com/mamoe/mirai/issues/2688

**Reply by 1298301390** [reply]
感谢答复，发消息没有报错，还有个疑问就是其他协议登录都提示禁止登录，这个要怎么解决吖

**Reply by ForteScarlet** [reply]
可参考 mirai 的 [第三方签名服务](https://mirai.mamoe.net/topic/2373/%E5%85%B3%E4%BA%8E%E7%AD%BE%E5%90%8D%E6%9C%8D%E5%8A%A1?lang=zh-CN) 自行搭建并使用。
第三方签名服务直接被mirai使用，不需要simbot层面的操作（最多就是在main方法一开始执行一些东西）

**Reply by 1298301390** [reply]
还有就是想问下，群消息时间如何在接收后拉起回复私聊消息呀，我看API里面没有这个方法
https://docs.simbot.forte.love/components/mirai/simbot-component-mirai-core/love.forte.simbot.component.mirai.event/-mirai-group-message-event/index.html

**Reply by ForteScarlet** [reply]
你可以在获取到此群成员的情况下（比如 `event.author` ）通过 `Member.send` 方法向一个成员发送私聊消息，或更稳妥的方式通过 `Bot.contact` 寻找对应的联系人（好友），然后得到好友对象后发送消息。
（好友消息比群临时消息更稳定，相对而言。）
